### ISSUE TITLE
Show a splash screen with loading progressbar on startup

### ISSUE THREAD

**Issue by nekohayo** [getting-things-gnome/gtg#43]
GTG is extremely slow to start, even if I prune old deleted tasks (thanks to the plugin that does that). My XML file weighs around 200 KB and I have around 400 tasks (90-100 actionable).

On a blazing fast Core i7 + SSD, it takes 15 seconds to start (warm start, not even cold start). On older computers, especially netbooks, you can easily wait 30 seconds.

I don't like to sweep big performance issues like that under the rug, but at least until you fix the startup time it would be really nice to have at least a splash screen, ideally with a progressbar, to indicate to the user that something is happening.

**Comment by izidormatusov** [comment]
Closing as duplicate of #109

**Comment by nekohayo** [comment]
Reopening/de-duplicating because until we have consistent sub-5 seconds startup / tree-building times "even with big datasets", this _could_ in theory still be worthwhile to have (especially if the app can autodetect that "this is a big file, it'll probably take me a while to process" and only show the splash+progress launch screen when needed... but of course if we end up implementing optimizations to the point where this is no longer relevant because startup performance is great even with thousands of tasks, then we can re-close this.

**Comment by aviwad** [comment]
So the code could go like

if (xml file exceeds certain size, show splash screen till ready)?

**Comment by pentux-GitHub** [comment]
Housekeeping pass, and this one can be settled with fresh numbers: the premise was a startup taking more than 5 seconds on big datasets. On current master the core loads the 1771-task bryce dataset in about 630ms (see #109 for the full measurements), and startup to a usable window feels fast. The splash screen no longer has a reason to exist. I'd think it's safe to close as obsolete.

**Comment by diegogangl** [comment]
Yup, it's always better to boot fast than to have splash screens 💪
